### PR TITLE
feat: add launchdarkly provider

### PR DESCRIPTION
**Pull request by chenrui333**

### CODE REVIEW

File: docs/launchdarkly.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+### Use with Launchdarkly
+
+Example:
+
+```
+export FASTLY_API_KEY=[FASTLY_API_KEY]
```
Comment (sergeylanzman) [inline docs/launchdarkly.md:6]: Typo

Comment (chenrui333) [reply docs/launchdarkly.md:6]: That is right!

**Comment by chenrui333** [comment]
Should be good now.

**Comment by chenrui333** [comment]
```
  providers/keycloak/generator.go:289:12: appendAssign: append result not assigned to the same slice (gocritic)
  		roles := append(clientRoles, realmRoles...)
  		         ^
  providers/datadog/datadog_provider.go:46:2: ifElseChain: rewrite if-else to switch statement (gocritic)
  	if args[3] != "" {
  	^
  providers/aws/rds.go:200: File is not `gofmt`-ed with `-s` (gofmt)
  		if (r.InstanceInfo.Type == "aws_db_instance" || r.InstanceInfo.Type == "aws_rds_cluster") {
  providers/aws/firehose.go:18: File is not `goimports`-ed (goimports)
  	"context"
  providers/aws/opsworks.go:18: File is not `goimports`-ed (goimports)
  	"context"
  providers/launchdarkly/launchdarkly_provider.go:24:6: type name will be used as launchdarkly.LaunchDarklyProvider by other packages, and that stutters; consider calling this Provider (golint)
  type LaunchDarklyProvider struct {
       ^
  providers/launchdarkly/launchdarkly_service.go:21:6: type name will be used as launchdarkly.LaunchDarklyService by other packages, and that stutters; consider calling this Service (golint)
  type LaunchDarklyService struct {
       ^
  providers/aws/route53.go:149:2: S1017: should replace this if statement with an unconditional strings.Replace (gosimple)
  	if strings.Contains(s, "\\052") {
  	^
```

**Comment by chenrui333** [comment]
should be good to go cc @sergeylanzman

**Pull request merged**